README.md

```diff
@@ -1,32 +1,355 @@
-# Lesson Title
+# Modifying Objects
 
 ## Learning Goals
 
-- Learning Goal 1
-- Learning Goal 2
+- Review the concept of reference types
+- Practice cloning objects
+- Practice modifying objects
+- Write pure functions that act on objects
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+So far we've learned how to access information from objects, both directly using
+dot or bracket notation, and dynamically using a `for...in` loop. In this
+lesson, we'll learn how to copy and modify objects. We'll also learn how to
+modify objects _nondestructively_, so we can keep our functions pure.
 
-## Topic 1
+## Review of Reference Types
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+You may recall that we learned in a previous lesson that arrays are what is
+known in JavaScript as _reference_ types, meaning that a reference to an array
+(e.g., a variable name) is pointing to a location in memory rather than the
+value of the array itself. This means that, if you copy an array using the
+assignment operator (`=`) and make a change to the original array, the same
+change will be made to the copy (and vice versa):
 
-## Topic 2
+```js
+myArray = ["one", "two", "three"];
+clone = myArray;
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+console.log(myArray);
+// => ["one", "two", "three"]
+console.log(clone);
+// => ["one", "two", "three"]
+
+myArray.push("four");
+
+console.log(myArray);
+// => ["one", "two", "three", "four"];
+console.log(clone);
+// => ["one", "two", "three", "four"];
+```
+
+Objects in JavaScript are also reference types. This isn't surprising given that
+in JavaScript, as we learned in an earlier lesson, arrays _are_ objects.
+
+## Cloning Objects
+
+Given that arrays are objects in JavaScript, you may wonder whether we can use
+the spread operator to copy objects as well as arrays. Well, we can! Let's take
+a look at an example:
+
+```js
+const disneyPrincesses = {
+  "The Little Mermaid": "Ariel",
+  Aladdin: "Jasmine",
+  Moana: "Moana",
+};
+```
+
+> Note that we aren't following JavaScript variable naming rules for our keys
+> here. That's okay, but remember the rules regarding using dot notation vs.
+> bracket notation. Also, remember that keys are case sensitive!
+
+Using the spread operator to create a copy that points to a different location
+in memory than the original is the same as with arrays, except using curly
+braces rather than square brackets:
+
+```js
+const princessClone = { ...disneyPrincesses };
+```
+
+That's it! Now we can make changes to our clone without changing the original
+object.
+
+But why is this important?
+
+You may recall that we talked about the advantages of using pure functions when
```

Comment:
Maybe a quick review on what those advantages are? Or at least a link to the section in the lesson where they learn this. Also, maybe just keep the focus on advantages in pure JS for now, as opposed to React. Mentioning it as a prominent advantage there could potentially lead them down React rabbit holes earlier than they should.